### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by nikasulo**
# Description
This PR contains the following functions as well as tests for them using JEST:

#### Caesar's Cypher
Caesar's cypher is a cryptographic cypher you can read all about [here](http://practicalcryptography.com/ciphers/caesar-cipher/)
Tests were written for the following edge cases:
* Shifts letters the appropriate number of times forward depending on the key given.
* Wraps around from z-a
* Maintains initial character case

#### Calculator (with addition, subtraction, division and multiplication functions)
Tests were written for the following edge cases: 
* Addition adds two numbers correctly
* Subtraction subtracts one number from another
* Multiplication gives the product of two numbers
* Division gives the result of one number divided by another

#### Analyze array
This function takes an array and returns the following details about the array: 
* Length
* Max(largest number)
* Min(smallest number)
* Average

#### Capitalize
A function that capitalizes a string
#### Reverse
A funtion that reverses a string

### CODE REVIEW

File: functions/analyse.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  for (let i = 0; i < array.length; i++){
+    if (array[i] < min) min = array[i];
+    if (array[i] > max) max = array[i];
+    average += array[i];
+  }
+  average = average/length
+  return {
+    min, max, average, length
+  }
+}
```
Comment (Oghenebrume50) [inline functions/analyse.js:15]: analyze should be an object that has function properties like min and max, so that it can be accessed like this analyze.min

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
STATUS: Changes Required

Hi @Oluwadamilareolusakin good job on this one man, :clap: you just make some few changes and we are good
The caeser chipher seems to change everything, this is not the desired behaviour you want to change only the alphabets and take note of punctuations they do not change
see the requirement
Please make changes and request a re-review 

Raphael Noriode

File: functions/analyse.js (hunk 1 above)
Comment (nikasulo) [reply functions/analyse.js:15]: Hey man, thanks for the review. Based on the description of the analyze function, it is meant to return an object that contains that properties of the array passed in as an argument which is what I've done :rocket:.

**Comment by nikasulo** [comment]
# Changes made as requested :smile: 
* I have corrected the issue with the punctuations
* The issue with the analyze function isn't really an issue, it works as described in the Odin Project Assignment

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @Oluwadamilareolusakin  :clap: 

**Great work making some necessary corrections from the previous review.**

**For some reason i ran into one failing test, screenshot below;**
image


**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by nikasulo** [comment]
@bellom a bit of oversight, I changed the way the program runs but forgot to uplate the tests. It's failing bexcuse punctuations are no longer encrypted. Will correct as requested.

**Comment by azdrenymeri** [comment]
### Tests are passing 👍 
#### You can merge
